src/lwc/rd2EntryForm/rd2EntryForm.js

```diff
@@ -40,120 +30,105 @@ import getSetting from '@salesforce/apex/RD2_entryFormController.getSetting';
+                this.isSettingReady = true;
+                this.isLoading = false;
+            });
     }
 
-    /*******************************************************************************
-    * @description Retrieve Recurring Donation Day of Month picklist values
+    /***
+    * @description Determine the parentId Sobject Type
     */
-    @wire(getPicklistValues, {fieldApiName: FIELD_DAY_OF_MONTH, recordTypeId: '$recurringDonationInfo.defaultRecordTypeId' })
-    wiredPicklistValues({error, data}) {
-       if(data) {
-          this.dayOfMonthPicklistValue = data.values;
-       }
-       if(error) {
-           this.handleError(error);
-       }
-    }
-
-   /*******************************************************************************
-   * @description Matches today's calender day with the RD Day of Month picklist value.
-   *    If the day does not match any picklist value(the case of 31), return api value 'Last_Day'
-   * @return String Today's day in Day of Month picklist api value
-   */
-    getTodayDateOfMonth() {
-        let currentDay = new Date().getDate().toString();
-       
-        let convertedPicklist = this.dayOfMonthPicklistValue.find(value => {
-                return value.value == currentDay;
-            });
+    handleParentIdType(parentSObjType) {
+        if (isNull(parentSObjType)) {
+            return;
+        }
 
-        return (convertedPicklist) 
-            ? convertedPicklist.value 
-            : this.lastDayPicklistValue;
+        if (parentSObjType === ACCOUNT_OBJECT.objectApiName) {
+            this.accountId = this.parentId;
+        } else if (parentSObjType === CONTACT_OBJECT.objectApiName) {
+            this.contactId = this.parentId;
+        }
     }
 
-    /*******************************************************************************
+    /***
     * @description Retrieve Recurring Donation Object info
     */
-    @wire(getObjectInfo, {objectApiName: RECURRING_DONATION_OBJECT.objectApiName })
-    wiredRDObjectInfo(response) {
+    @wire(getObjectInfo, { objectApiName: RECURRING_DONATION_OBJECT.objectApiName })
+    wiredRecurringDonationObjectInfo(response) {
         if (response.data) {
-            this.recurringDonationInfo = response.data;
-            this.setFields(this.recurringDonationInfo.fields);
+            let rdObjectInfo = response.data;
+            this.setFields(rdObjectInfo.fields);
             this.fieldInfos = this.buildFieldDescribes(
-                this.recurringDonationInfo.fields,
-                this.recurringDonationInfo.apiName);
-                this.isFieldInfoReady = true;
+                rdObjectInfo.fields,
+                rdObjectInfo.apiName
+            );
+
+            if (isNull(this.recordId)) {
+                this.isRecordReady = true;
+            }
         }
 
         if (response.error) {
-            this.isFieldInfoReady = true;
-            handleError(response.error);
+            this.isRecordReady = true;
```

Comment:
Lesson learned: do not set `this.isRecordReady` outside of the `response.data` and `response.error` blocks. Moreover, errors are generated and displayed in the console log when the RD tab is displayed even before clicking on the New button.